src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -0,0 +1,555 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.Operations
+{
+    internal sealed class ControlFlowGraphBuilder : OperationCloner
+    {
+        private readonly BasicBlock _entry = new BasicBlock(BasicBlockKind.Entry);
+        private readonly BasicBlock _exit = new BasicBlock(BasicBlockKind.Exit);
+        private ArrayBuilder<BasicBlock> _blocks;
+        private BasicBlock _currentBasicBlock;
+        private IOperation _currentStatement;
+        private ArrayBuilder<IOperation> _evalStack;
+        private int _availableCaptureId;
+
+        private ControlFlowGraphBuilder()
+        { }
+
+        public static ImmutableArray<BasicBlock> Create(IBlockOperation body)
+        {
+            var builder = new ControlFlowGraphBuilder();
+            var blocks = ArrayBuilder<BasicBlock>.GetInstance();
+            builder._blocks = blocks;
+            builder._evalStack = ArrayBuilder<IOperation>.GetInstance();
+            blocks.Add(builder._entry);
+
+            builder.VisitStatement(body);
+            builder.AppendNewBlock(builder._exit);
+
+            // Do a pass to eliminate blocks without statements and only Next set.
```

Comment:
> and only Next set.
Is that still reflecting the logic of `Pack`? Maybe this part of the comment can be removed.